hydxlib/threedi.py

```diff
@@ -420,10 +420,26 @@ def get_discharge_coefficients(self, hydx_connection, hydx_structure):
             or hydx_connection.stromingsrichting == "2_1"
         ):
             hydx_connection.discharge_coefficient_negative = (
-                hydx_structure.afvoercoefficientoverstortdrempel
+                self.get_structure_specific_coefficient(hydx_structure)
             )
         return hydx_connection
 
+    def get_structure_specific_coefficient(self, hydx_structure):
+        if hydx_structure.typekunstwerk == 'OVS':
```

Comment:
Deze code is prima.
Kan het nog duidelijker door (gewoon in deze methode) een mapping neer te zetten?

    mapping = {"OVS": hydx....drempel,
                       "DRL": hydfsdfkjlsdfkljsdf}
    if not hydx_structure.typekunstwerk in mapping:
         logger.gil-als-speenvarken()
        return None
    return mapping[hydx_ssdddfsdftypekunstwerk]

Hangt er van af hoeveel er komen. En of je misschien nog wat extra's uit moet rekenen bij één van de varianten.

Reply:
Erg goed idee!

Ik heb het verwerkt.
Is dit wat je bedoeld?
(de foutmelding vind ik stiekem niet zo nodig, aangezien het onwaarschijnlijk is dat er iets anders gebeurd in de code, dat zou duiden op een programmeursfout, niet op een data fout.)